packages/block-library/src/embed/edit.js

```diff
@@ -52,12 +59,36 @@ export function getEmbedEditComponent( title, icon, responsive = true ) {
 			}
 		}
 
+		handlePaste() {
+			const { url } = this.props.attributes;
+			const { fetching, cannotEmbed } = this.props;
+
+			// This URL can't be embedded, so replace with the fallback block.
+			if ( url && ! fetching && cannotEmbed ) {
+				this.props.onReplace(
+					createPasteFallbackBlock( this.props.attributes.url )
+				);
+				return;
+			}
+
+			// This URL can be embedded, so pass it over to the default embed block to continue.
+			if ( url && ! fetching && ! cannotEmbed ) {
+				this.props.onReplace(
+					createBlock( DEFAULT_EMBED_BLOCK, { url } )
+				);
+			}
+		}
+
 		componentDidUpdate( prevProps ) {
 			const hasPreview = undefined !== this.props.preview;
 			const hadPreview = undefined !== prevProps.preview;
 			const switchedPreview = this.props.preview && this.props.attributes.url !== prevProps.attributes.url;
 			const switchedURL = this.props.attributes.url !== prevProps.attributes.url;
 
+			if ( pasting ) {
```

Comment:
Is this really necessary if we call the past handler in the constructor? Also it feels like `componentDidMount` is a better option for this call.

Reply:
Ok so I'm not sure I understand everything here but it doesn't work if it's only on mount

Reply:
Yeah the problem is that to work out if something is embeddable, we need the response from the embed proxy API, because WordPress embeds don't have a regex that can match them, and you can paste, for example, a YouTube URL which matches the embed provider's regex, but isn't actually embeddable. So we need to do this when the component updates the preview.

Reply:
This could be solved using a dedicated `pasteEmbed` action instead of handling it on the component level.

Reply:
Yeah, that would need a bit of a refactor so that the `withSelect` part was in common code, but it's doable, and definitely would make this cleaner.